### PR TITLE
get the async example working

### PR DESCRIPTION
**Pull request by Rob-Johnson**
also make the event handler a namedtuple

### CODE REVIEW

File: task_processing/runners/async.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
+from collections import namedtuple
 from threading import Thread
 
 from six.moves.queue import Empty
 
 from task_processing.interfaces.runner import Runner
 
+EventHandler = namedtuple('EventHandler', ['matcher_func', 'cb'])
```
Comment (keymone) [inline task_processing/runners/async.py]: maybe `predicate` instead of `matcher_func`?

File: task_processing/plugins/mesos/translator.py [hunk 2]
```diff
@@ -2,17 +2,17 @@
 
 # https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto
 MESOS_TASK_STATUS_TO_EVENT = {
-    0: Event(platform_type='starting'),
-    1: Event(platform_type='running'),
-    2: Event(platform_type='finished'),
-    3: Event(platform_type='failed'),
+    0: Event(platform_type='starting', terminal=False),
```
Comment (keymone) [inline task_processing/plugins/mesos/translator.py]: make `initial=False` in field definition

File: examples/async.py [hunk 3]
```diff
@@ -16,28 +16,29 @@ def __init__(self):
...
     mesos_address = os.environ['MESOS']
     executor = MesosExecutor(
-        credential_secret_file="/src/task_processing/examples/cluster/secret",
+        credential_secret_file="/src/examples/cluster/secret",
         mesos_address=mesos_address
     )
 
     counter = Counter()
-    runner = Async(executor, [(EventTerminal, counter.process_event)])
+    runner = Async(executor, [EventHandler(matcher_func=lambda x: x.terminal, cb=counter.process_event)])
```
Comment (keymone) [inline examples/async.py]: 80 chars

File: task_processing/plugins/mesos/translator.py (hunk 2 above)
Comment (Rob-Johnson) [inline task_processing/plugins/mesos/translator.py]: why? I don't think there is a reasonable default - I'd leave it to the implementer to decide either way per Event?

Comment (keymone) [inline task_processing/plugins/mesos/translator.py]: should `None` mean anything different from `False` then? if it does, we need to make sure an error is thrown when such event ends up in `if event.terminal` (make an object with __bool__ method that throws an exception). if it doesn't - then let's at least make the field mandatory?

Comment (Rob-Johnson) [inline task_processing/plugins/mesos/translator.py]: | should None mean anything different from False then?

You shouldn't be able to do ``None``? maybe I can provide a predicate like: ```lambda x: x in ['True', 'False']``` if the typing isn't strong enough.

|  if it doesn't - then let's at least make the field mandatory?
agreed

**Comment by keymone** [comment]
ship.

**Pull request merged**